### PR TITLE
FOGL-1389 - Added feature to enable/disable schedule by schedule update vide PUT

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
This PR adds below features:
-  We can now create a new schedule with "enabled" field set to "t" (or "f") and appropriate action (enable schedule in case of "t") will be initiated.
- Similarly, now any schedule can be enabled/disabled vide PUT verb with {"enabled": "t"/"f"} payload.
- Now we can create a configuration key for a service prior to its induction in the foglamp server and it will be added to the service whenever the service is added vide add_service() call. Please refer to below example.

we can create a configuration key "simplesima" even if no simplesima service is added to foglamp.

`curl -d '{"key": "simplesima", "description": "simple simulator A", "value": {"info": {"description": "Test", "type": "boolean", "default": "true"}}}' -X POST http://localhost:8081/foglamp/category`

we can find the key vide:
```
curl -X GET http://localhost:8081/foglamp/category | jq

{
  "categories": [
    ...
    ...
    {
      "description": "simple simulator A",
      "key": "simplesima"
    }
  ]
}
```
and if we examine the simplesima key value in the DB configuration table , we will find it to be:
```
{
	"info": {
		"description": "Test",
		"type": "boolean",
		"value": "true",
		"default": "true"
	}
}
```
Now we add the service simplesima vide:
`curl -X POST http://localhost:8081/foglamp/service -d '{"name": "simplesima", "type": "south", "plugin": "simplesima", "enabled": "t"}'
`

and curl -X GET http://localhost:8081/foglamp/category | jq gives
```
{
  "categories": [
   ...
   ...
    {
      "description": "simplesima service configuration",
      "key": "simplesima"
    }
  ]
}

```

and if we examine the simplesima key value in the DB configuration table , we will find it to be:
```
{
  "plugin": {
    "value": "simplesima",
    "default": "simplesima",
    "description": "Simple Simulator async plugin",
    "type": "string"
  },
  "info": {
    "value": "true",
    "default": "true",
    "description": "Test",
    "type": "boolean"
  },
  "management_host": {
    "value": "127.0.0.1",
    "default": "127.0.0.1",
    "description": "Management host",
    "type": "string"
  }
}
```

### CODE REVIEW

File: python/foglamp/services/core/api/service.py [hunk 1]
```diff
@@ -136,7 +142,7 @@ def get_service_records():
         schedule.exclusive = True
         schedule.enabled = False
         # Save schedule
-        await server.Server.scheduler.save_schedule(schedule)
+        await server.Server.scheduler.save_schedule(schedule, )
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/service.py]: ?

File: python/foglamp/services/core/api/service.py [hunk 2]
```diff
@@ -87,6 +89,10 @@ def get_service_records():
             raise web.HTTPBadRequest(reason='Missing type property in payload.')
         if not service_type in ['south', 'north']:
             raise web.HTTPBadRequest(reason='Only north and south types are supported.')
+        if enabled is not None:
+            if enabled not in ['t', 'f', 'true', 'false']:
+                raise web.HTTPBadRequest(reason='Only "t", "f", "true", "false" are allowed for value of enabled.')
+        enabled_changed = True if enabled.lower() in ['t', 'true'] else False
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/service.py]: is_enabled  is better name then it should be mapped to is_schedule_enabled when plugged with scheduler

File: python/foglamp/services/core/scheduler/scheduler.py [hunk 3]
```diff
@@ -1116,6 +1116,24 @@ def _schedule_row_to_schedule(cls,
...
+            disable_schedule() will return unconditionally without doing any action otherwise suitable action will be
+            invoked. This implies that if we want to properly change any attribute of a running schedule, it must
+            first be disable, attribtue value changed and then it should be restarted. Othrwise schedule will continue
+            to run with unpredictable behaviour.
+
+            For a new schedule, if enabled is set to True, the schedule will be enabled otherwise disable_schedule()
+            will be called and which will return unconditionally without any action.
+        """
+        if enabled_changed is not None:
+            if enabled_changed is True:
```
Comment (praveen-garg) [inline python/foglamp/services/core/scheduler/scheduler.py]: if is_schedule_enabled:

**Comment by praveen-garg** [comment]
@amarendra-dianomic  unit tests result?

**Comment by amarendra-dianomic** [comment]
@MarkRiddoch @praveen-garg the PR has been updated per feedback and also as per clarity provided  at [FOGL-1389](https://scaledb.atlassian.net/browse/FOGL-1389?focusedCommentId=39123&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-39123). Please check.

**Pull request merged**